hspec-core/test/Test/Hspec/Core/Config/OptionsSpec.hs

```diff
@@ -115,10 +115,11 @@ spec = do
         fromLeft (parseOptions [("~/.hspec", ["--fail"])] Nothing [] []) `shouldBe` (ExitFailure 1,
           unlines [
             "my-spec: option `--fail' is ambiguous; could be one of:"
-          , "    --fail-on-focused      fail on focused spec items"
-          , "    --fail-on-pending      fail on pending spec items"
-          , "    --fail-fast            abort on first failure"
-          , "    --failure-report=FILE  read/write a failure report for use with --rerun"
+          , "    --fail-on-focused            fail on focused spec items"
+          , "    --fail-on-pending            fail on pending spec items"
+          , "    --fail-on-empty-description  fail on spec items with empty descriptions"
+          , "    --fail-fast                  abort on first failure"
```

Comment:
This seems pretty reasonable!

I think my main thought here is that - we've got a bunch of `--fail-on-*` options. Possibly it's time to write `--fail-on` as a flag that can be repeated multiple times with an enum argument, like,

```
my-tests --fail-on first --fail-on focused --fail-on pending --fail-on empty-description
```

This reduces the surface area of new CLI arguments and can make it easier to document the behavior - instead of N CLI flags, you have 1 CLI flag with an enum of N constructors.

Reply:
I like that idea. Maybe even allow comma separated lists, e.g.
```
spec --fail-on=pending,focused,empty-description
```
We should give that a try eventually.